subprojects/core-api/src/main/java/org/gradle/api/artifacts/result/ResolvedComponentResult.java

```diff
@@ -91,16 +92,28 @@
      */
     List<ResolvedVariantResult> getVariants();
 
+    /**
+     * Returns all the variants for this component, even ones that weren't selected.
+     *
+     * @return all variants for this component
+     * @since 7.5
+     */
+    @Incubating
+    List<ResolvedVariantResult> getAllVariants();
```

Comment:
`ResolvedComponentResult` can be used as a task input. This new state should be added to the snapshotting. Tests for this are in `DependencyManagementResultsAsInputsIntegrationTest` and snapshotting support is in `ResolvedComponentResultSerializer`.